applications/zklogin-Reclaim-protocol.md

```diff
@@ -0,0 +1,299 @@
+
+Additionally, after speaking with the members of the Polkadot Foundation, we have been able to identify zklogin and [epassport](https://github.com/w3f/Grants-Program/blob/master/docs/RFPs/epassport-zk-validation.md) to be important use cases for Dapp builders and users that can be implemented uniquely through Reclaim Protocol. Through Reclaim, users can not only validate their login credentials but also provide verifiable proofs of various personal attributes and activities. **For instance, they can prove the number of rides taken on Uber, expenditures on Amazon, or their scores on Chess.com, thereby establishing a robust proof of personhood which is instrumental in combating fraudulent actors and bots. Such proof of personhood can now be made available on-chain via Reclaim protocol within the Polkadot ecosystem.**
+
+We are a small team (20+ members) with finite resources and want to collaborate with ecosystems that demonstrate a genuine commitment to enable use cases that are likely to scale to a billion users. Our team comprises primarily engineers who have previously worked at Meta, Microsoft, Google and have contributed to various web3 open source projects. Given Polkadot's strong emphasis on streamlining user onboarding,  user experience, and implementing bot protection, coupled with our team's expertise and background, we believe that our team is one of the most suited to realise Polkadot's vision of building zk login and implementing Proof of Personhood for Dapps on top of Polkadot.
+
+
+
+### Project Details
+
+[zk Login Overview](https://questbook.notion.site/Implementing-zklogin-and-Proof-of-Personhood-using-Reclaim-Protocol-6a1d31a7d70c46c482682742abbbe0cb#:~:text=using%20Reclaim%20Protocol-,zk%20Login%20Overview,-Reclaim%20Protocol%20Overview)
+
+**Reclaim Protocol Technical Deep Dive**
+
+Reclaim empowers users to generate Zero-Knowledge Proofs for any online user profile. To generate a claim, users first need to log into the relevant website. This login process, involving an HTTPS request and its subsequent response, is channeled through an HTTPS Proxy Server known as an 'attestor'. This attestor oversees the encrypted data exchange between the user and the website. Subsequently, users provide keys that disclose non-sensitive parts of the request to the attestor. With this, the attestor can view the request in its entirety, barring confidential details like authentication data, and can confirm its legitimacy.
```

Comment:
An online user profile at some website not otherwise linked to polkadot?  

Is this HTTPS Proxy Server learning the user's password or otherwise being an MitM?  Is it run by the user themselves or someone else?

Reply:
Indeed, online profile of users on various centralized platforms such as Facebook, Instagram, Uber, and others, which are not inherently linked to Polkadot to bridge the gap between centralized platforms and Dapps on top of Polkadot.



We have discussed solutions to potential attacks including MITM  against Reclaim protocol in Appendix D of the whitepaper. Sharing the relevant excerpt from the [whitepaper](https://drive.google.com/file/d/1wmfdtIGPaN9uJBI1DHqN903tP9c_aTG2/view) for easy access.  

![Screenshot 2024-01-10 at 12 56 57 PM](https://github.com/pioneersprize/Polkadot-Pioneers-Prize/assets/83450876/1b2d63c2-6baa-4548-80c6-31f52d7d4bde)

As of today, the nodes are run by us and there is a trust assumption on us. This trust assumption is however not on privacy. Though technically the solution is that of an HTTP Proxy, aka MITM, it doesn’t allude to privacy concerns. That is because the data is encrypted end to end by TLS. The client will be trivially able to detect an MITM attack if any via a certificate mismatch error on their browser. We have also discussed a [more sophisticated attack here](https://blog.reclaimprotocol.org/posts/fake-key-reveal/) and its mitigation.

We are actively in the process of decentralizing this attestor. The requirement to become an attestor is to stake some tokens. The detailed description can be found in [section 4](https://drive.google.com/file/d/1wmfdtIGPaN9uJBI1DHqN903tP9c_aTG2/view) of the whitepaper. Additionally, we have gotten our architecture and circuits audited (https://www.zksecurity.xyz/blog/posts/reclaim/), and more modules are under audit right now. The open-source code can be found here - https://github.com/reclaimprotocol